src/t_stream.c

```diff
@@ -2456,22 +2462,22 @@ NULL
             addReply(c,shared.czero);
         }
     } else if (!strcasecmp(opt,"CREATECONSUMER") && c->argc == 5) {
-        int created = 0;
-        streamLookupConsumer(cg,c->argv[4]->ptr,SLC_NOREFRESH,&created);
-        if (created) {
+        streamConsumer *created = streamCreateConsumer(cg,c->argv[4]->ptr,
+                                           c->argv[2],c->db->id);
+        addReplyLongLong(c,created ? 1 : 0);
+    } else if (!strcasecmp(opt,"DELCONSUMER") && c->argc == 5) {
+        long long pending = 0;
+        streamConsumer *consumer = streamLookupConsumer(cg,c->argv[4]->ptr,0);
+        if (consumer) {
+            /* Delete the consumer and returns the number of pending messages
+             * that were yet associated with such a consumer. */
+            pending = raxSize(consumer->pel);
+            streamDelConsumer(cg,c->argv[4]->ptr);
```

Comment:
1. I refactored streamDelConsumer() because when deleting a non-existent consumer, it will always notify and dirty++. I think this is a bug. (maybe I should mention a new PR)
2. I didn't have the keyspace notification and dirty++ inside streamDelConsumer() because streamDelConsumer() was only called at this place, I just kept it as it is.
3. I will update the comment, thanks for reminding.

Reply:
please also update the top comment of the PR (we'll refer to it in the release notes).
p.s. review will be slightly easier if you push additional commits rather than amend the first one.
we'll later squash them when merging and use the PR top comment as the commit comment